java/src/com/ibm/streamsx/topology/internal/streams/InvokeCancel.java

```diff
@@ -18,10 +18,17 @@
     static final Logger trace = Topology.STREAMS_LOGGER;
 
     private final BigInteger jobId;
+    private final String userName;
 
+    public InvokeCancel(BigInteger jobId, String userName) {
+        super();
+        this.jobId = jobId;
+        this.userName = userName;
+    }
     public InvokeCancel(BigInteger jobId) {
         super();
         this.jobId = jobId;
+        this.userName = "";
```

Comment:
It's usual in Java to have `null` represent an unset value, not the empty string.